content/articles/understanding-icomparable-and-icomparer-interface-in-c#/index.md

```diff
@@ -0,0 +1,288 @@
+### Introduction
+In this article we will be talking about the IComparable and IComparer interfaces. The two interfaces may sound similar but they all perform different activities. You can use the interfaces together or separately  but they work best when they are combined together for sorting and also comparison purposes.
+
+We will briefly talk about sorting then dive into the comparison interfaces 
+
+### Table of contents
+- [An overview of sorting](#an-overview-of-sorting)
+- [Getting started with the IComparer interface  and IComparable Interface](#getting-started-with-the-icomparer-and-icomparable-interface)
+- [Implementing IComparable and IComparer Interfaces](#implementing-icomparable-and-icomparer-interfaces)
+- [Difference between the IComparable and IComparer Interfaces](#difference-between-the-icomparable-and-icomparer-interfaces)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+### An overview of sorting
+Let's start with the basics of sorting. Let's assume that we wish to sort a list of integers ascendingly. The Sort function offered by Microsoft can be used because we are utilizing Int and the built-in List.
+
+In List <T>.sort, the method of sorting parameter is a comparator, with a couple of overloads.
+
+1. Sort() which uses the default comparer to sort the elements in the entire [List<T>](https://www.tutorialsteacher.com/csharp/csharp-list). It sorts only elements of common type like INT and STRING as shown below.

```

Comment:
To point out a piece of code use this `Sort()`. Repeat this with all the small code snippets.